robocode.battle/src/main/java/net/sf/robocode/recording/RecordManager.java

```diff
@@ -148,64 +148,58 @@ ITurnSnapshot readSnapshot() {
     }
 
     public void loadRecord(String recordFilename, BattleRecordFormat format) {
-        FileInputStream fis = null;
-        BufferedInputStream bis = null;
-        ZipInputStream zis = null;
-        ObjectInputStream ois = null;
-        InputStream xis = null;
-
-        FileOutputStream fos = null;
-        BufferedOutputStream bos = null;
-        ObjectOutputStream oos = null;
-
-        try {
-            createTempFile();
-            fis = new FileInputStream(recordFilename);
-            bis = new BufferedInputStream(fis, 1024 * 1024);
-
+        createTempFile();
+
+        try (
+                FileInputStream fis = new FileInputStream(recordFilename);
+                BufferedInputStream bis = new BufferedInputStream(fis, 1024 * 1024);
+                ZipInputStream zis = (format == BattleRecordFormat.BINARY_ZIP || format == BattleRecordFormat.XML_ZIP) ? new ZipInputStream(bis) : null;
+                FileOutputStream fos = (format == BattleRecordFormat.BINARY || format == BattleRecordFormat.BINARY_ZIP || format == BattleRecordFormat.XML_ZIP || format == BattleRecordFormat.XML) ? new FileOutputStream(tempFile) : null;
+                BufferedOutputStream bos = (fos != null) ? new BufferedOutputStream(fos, 1024 * 1024) : null;
+                ObjectOutputStream oos = (bos != null && (format == BattleRecordFormat.BINARY || format == BattleRecordFormat.BINARY_ZIP)) ? new ObjectOutputStream(bos) : null
+        ) {
+            // Extracted ternary operation for ObjectInputStream initialization
+            ObjectInputStream ois = null;
```

Comment:
I am not sure about this part. Maybe we can divide this part into several try blocks under specific if conditions.

ois may also need to be defined in the try-with-resource to close

Reply:
Creating multiple if blocks will create code repetition as the conditional statement are used to use the same snippet of code.

Adding OIS in the try statement generates a new technical debt of "Nested statements" and to resolve that we would have to write it as above.